processing/src/main/java/org/apache/druid/query/topn/TopNQueryQueryToolChest.java

```diff
@@ -292,19 +292,13 @@ public byte[] computeCacheKey(TopNQuery query)
       {
         final CacheKeyBuilder builder = new CacheKeyBuilder(TOPN_QUERY)
             .appendCacheable(query.getDimensionSpec())
-            .appendCacheable(query.getTopNMetricSpec())
             .appendInt(query.getThreshold())
             .appendCacheable(query.getGranularity())
             .appendCacheable(query.getDimensionsFilter())
             .appendCacheables(query.getAggregatorSpecs())
-            .appendCacheable(query.getVirtualColumns());
-
-        final List<PostAggregator> postAggregators = prunePostAggregators(query);
-        if (!postAggregators.isEmpty()) {
-          // Append post aggregators only when they are used as sort keys.
-          // Note that appending an empty list produces a different cache key from not appending it.
-          builder.appendCacheablesIgnoringOrder(postAggregators);
-        }
+            .appendCacheable(query.getVirtualColumns())
```

Comment:
Similar issue to the one with groupBy (postaggs refer to dim/agg/other-postagg by name).